File: JavaScript Engine/ai/action.js
export const Actions = ['buy', 'sell', 'do_nothing'];
export const Total_Actions = 3;
export class Action {
    constructor(action_type) {
        if (!Actions.includes(action_type)) {
            throw new Error(`Invalid action type: ${action_type}`);
        }
        this.action_type = action_type;
        this.action_index = Actions.indexOf(action_type);
    }

    toString() {
        return this.action_type;
    }
}

File: classes/DQAgent/action.py
Actions = ['buy', 'sell', 'do_nothing']
Total_Actions = len(Actions)
Action_Size = 1
class Action:
    def __init__(self, action_type: str):
        self.action_type = action_type
        self.action_index = Actions.index(action_type)
    def __str__(self):
        return self.action_type